src/TimeSteppers/TimeSteppers.jl

```diff
@@ -0,0 +1,187 @@
+module TimeSteppers
+
+export 
+    AdamsBashforthTimeStepper,
+    time_step!, 
+    compute_w_from_continuity!
+
+using Oceananigans: device
+
+using GPUifyLoops: @launch, @loop, @unroll
+
+import Oceananigans: TimeStepper
+
+using Oceananigans: AbstractGrid, Model, Tendencies, tracernames, 
+                    @hascuda, CPU, GPU, launch_config, datatuples, datatuple,
+                    @loop_xyz,
+
+                    buoyancy_perturbation,
+                    x_f_cross_U, y_f_cross_U, z_f_cross_U,
+
+                    fill_halo_regions!, apply_z_bcs!, solve_poisson_3d!, PoissonBCs, PPN, PNN,
+
+                    run_diagnostic, write_output, time_to_run
+
+@hascuda using CUDAnative, CUDAdrv, CuArrays
+
+using ..Operators
+
+using ..TurbulenceClosures: ∂ⱼ_2ν_Σ₁ⱼ, ∂ⱼ_2ν_Σ₂ⱼ, ∂ⱼ_2ν_Σ₃ⱼ, ∇_κ_∇c,
+                            calculate_diffusivities!, ▶z_aaf
+
+const Tx = 16 # CUDA threads per x-block
+const Ty = 16 # CUDA threads per y-block
+
+"""
+    TimeStepper(name, args...)
+
+Returns a timestepper with name `name`, instantiated with `args...`.
+
+Example
+=======
+
+julia> stepper = TimeStepper(:AdamsBashforth, Float64, CPU(), grid, tracernames)
+"""
+function TimeStepper(name::Symbol, args...)
+    fullname = Symbol(name, :TimeStepper)
+    return eval(Expr(:call, fullname, args...))
+end
+
+# Fallback
+TimeStepper(stepper, args...) = stepper
+
+"""Returns the arguments passed to boundary conditions functions."""
+boundary_condition_function_arguments(model) = (model.clock.time, model.clock.iteration, datatuple(model.velocities), 
+                                                datatuple(model.tracers), model.parameters)
+
+"""
+    time_step!(model; Nt, Δt, kwargs...)
+
+Step forward `model` `Nt` time steps with step size `Δt`.
+
+The kwargs are passed to the `time_step!` function specific to `model.timestepper`.
+"""
+time_step!(model; Nt, Δt, kwargs...) = time_step!(model, Nt, Δt; kwargs...)
+
+function time_step!(model, Nt, Δt; kwargs...)
+
+    if model.clock.iteration == 0
+        [ run_diagnostic(model, diag) for diag in values(model.diagnostics) ]
+        [ write_output(model, out)    for out  in values(model.output_writers) ]
+    end
+
+    for n in 1:Nt
+        time_step!(model, Δt; kwargs...)
+
+        [ time_to_run(model.clock, diag) && run_diagnostic(model, diag) for diag in values(model.diagnostics) ]
+        [ time_to_run(model.clock, out) && write_output(model, out) for out in values(model.output_writers) ]
+    end
+
+    return nothing
+end
+
```

Comment:
Maybe add a banner like
```julia
####
#### Time-stepping kernels/function that are independent of the TimeStepper
####
```
as the individual steps are now split between multiple files.

Reply:
Done.